project/stats/views.py

```diff
@@ -0,0 +1,81 @@
+from django.shortcuts import render
+from datetime import date, timedelta
+from accounts.models import Profile
+from tickets.models import Ticket
+
+
+def get_all_stats(request):
+    """ Filter all necessary objects to render Statistics using Chart.js """
+
+    """ ----- USERS ----- """
+    # top 3 donors:
+    users_top_three = Profile.objects.all().order_by("-total_donated")[:3]
+
+    """ ----- MONTHLY / WEEKLY / DAILY ----- """
+    date_search_start = date.today() - timedelta(days=30)
+    date_search_end = date.today()
+
+    """ ----- FEATURES ----- """
+    # top 5 most voted
+    features_top_five = Ticket.objects.filter(
+        ticket_type="2").order_by("-upvotes")[:5]
+    # total features
+    features_total = Ticket.objects.filter(
```

Comment:
we are making many database queries to get these counts, you can make this more efficient.

Reply:
managed to get the 17 different db queries down to 3 queries with same result - will be on next push